examples/util/WebNNRunner.js

```diff
@@ -0,0 +1,427 @@
+   */
+  setEagerMode = (flag) => {
+    this._bEagerMode = flag;
+  };
+
+  /**
+   * This method is to set '_supportedOps'.
+   * @param ops: A string for supported ops.
+   */
+  setSupportedOps = (ops) => {
+    this._supportedOps = ops;
+  };
+
+  /**
+   * This method is to set '_rawModel'.
+   * @param model: A string for model.
+   */
+  _setRawModel = (model) => {
+    this._rawModel = model;
+  };
+
+  /**
+   * This method is to set '_subgraphsSummary'.
+   * @param summary: A string for summary.
+   */
+  _setSubgraphsSummary = (summary) => {
+    this._subgraphsSummary = summary;
+  };
+
+  /**
+   * This method is to set '_modelRequiredOps'.
+   * @param ops: A string for required ops of model.
+   */
+  _setModelRequiredOps = (ops) => {
+    this._modelRequiredOps = ops;
+  };
+
+  /**
+   * This method is to set '_deQuantizeParams'.
+   * @param params: A string for deQuantize params.
+   */
+  _setDeQuantizeParams = (params) => {
+    this._deQuantizeParams = params;
+  };
+
+  /**
+   * This method is to load model file with specified url.
+   * @param url: A string for model file url.
+   * @returns {string} This returns status of load model file, ['ERROR' | 'SUCCESS'].
+   */
+  _loadModelFile = async (url) => {
+    let status = 'ERROR';
+    let rawModel = null;
+
+    if (url !== undefined) {
+      const arrayBuffer = await this._loadURL(url, this._progressHandler, true);
+      const bytes = new Uint8Array(arrayBuffer);
+      switch (url.split('.').pop()) {
+        case 'tflite':
+          const flatBuffer = new flatbuffers.ByteBuffer(bytes);
+          rawModel = tflite.Model.getRootAsModel(flatBuffer);
+          rawModel._rawFormat = 'TFLITE';
+          status = 'SUCCESS'
+          printTfLiteModel(rawModel);
+          break;
+        case 'onnx':
+          const err = onnx.ModelProto.verify(bytes);
+          if (err) {
+            throw new Error(`The model file ${url} is invalid, ${err}`);
+          }
+          rawModel = onnx.ModelProto.decode(bytes);
+          rawModel._rawFormat = 'ONNX';
+          status = 'SUCCESS'
+          printOnnxModel(rawModel);
+          break;
+        case 'bin':
+          const networkFile = url.replace(/bin$/, 'xml');
+          const networkText = await this._loadURL(networkFile);
+          const weightsBuffer = bytes.buffer;
+          rawModel = new OpenVINOModel(networkText, weightsBuffer);
+          rawModel._rawFormat = 'OPENVINO';
+          status = 'SUCCESS';
+          break;
+        default:
+          throw new Error(`Unrecognized model format, support TFLite | ONNX | OpenVINO model`);
+      }
+    } else {
+      throw new Error(`There's none model file info, please check config info of modelZoo.`);
+    }
+
+    this._setRawModel(rawModel);
+    this._setLoadedFlag(true);
+    return status;
+  };
+
+  /**
+   * This method is to load model with given model info.
+   * @param modelInfo: A string for model info.
```

Comment:
Is the model info a string? Please refer to the class definition. And I prefer to having that in RunnerBase. So every caller and implementation knows the definition.